cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageGenerator.java

```diff
@@ -49,40 +59,54 @@
     };
 
   private final LineageStore lineageStore;
+  private final Store store;
 
   @Inject
-  LineageService(LineageStore lineageStore) {
+  LineageGenerator(LineageStore lineageStore, Store store) {
     this.lineageStore = lineageStore;
+    this.store = store;
   }
 
   /**
    * Computes lineage for a dataset between given time period.
    *
    * @param sourceDataset dataset to compute lineage for
-   * @param start start time period
-   * @param end end time period
+   * @param startMillis start time period
+   * @param endMillis end time period
    * @param levels number of levels to compute lineage for
    * @return lineage for sourceDataset
    */
-  public Lineage computeLineage(final Id.DatasetInstance sourceDataset, long start, long end, int levels) {
-    return doComputeLineage(sourceDataset, start, end, levels);
+  public Lineage computeLineage(final Id.DatasetInstance sourceDataset, long startMillis, long endMillis, int levels) {
+    return doComputeLineage(sourceDataset, startMillis, endMillis, levels);
   }
 
   /**
    * Computes lineage for a stream between given time period.
    *
    * @param sourceStream stream to compute lineage for
-   * @param start start time period
-   * @param end end time period
+   * @param startMillis start time period
+   * @param endMillis end time period
    * @param levels number of levels to compute lineage for
    * @return lineage for sourceStream
    */
-  public Lineage computeLineage(final Id.Stream sourceStream, long start, long end, int levels) {
-    return doComputeLineage(sourceStream, start, end, levels);
+  public Lineage computeLineage(final Id.Stream sourceStream, long startMillis, long endMillis, int levels) {
+    return doComputeLineage(sourceStream, startMillis, endMillis, levels);
   }
 
-  private Lineage doComputeLineage(final Id.NamespacedId sourceData, long start, long end, int levels) {
-    LOG.trace("Computing lineage for data {}, start {}, end {}, levels {}", sourceData, start, end, levels);
+  private Lineage doComputeLineage(final Id.NamespacedId sourceData, long startMillis, long endMillis, int levels) {
+    LOG.trace("Computing lineage for data {}, startMillis {}, endMillis {}, levels {}",
+              sourceData, startMillis, endMillis, levels);
+
+    // Convert start time and end time period into scan keys in terms of program start times.
+    Set<RunId> runningInRange = store.getRunningInRange(TimeUnit.MILLISECONDS.toSeconds(startMillis),
+                                                        TimeUnit.MILLISECONDS.toSeconds(endMillis));
+    if (LOG.isTraceEnabled()) {
```

Comment:
I am printing the size of the container in the log message, so thought it safe to put a check - in case size is not a constant time function.